### PR TITLE
#5153 Add general IT tests for settings dialog

### PR DESCRIPTION
**Pull request by saivirtue**
### Checklist
- [x] [Review Contribution Guidelines](https://ohara.readthedocs.io/en/latest/contributing.html)
- [x] Add Reviewers (see [Authors](https://github.com/oharastream/ohara#ohara-team))
- [x] Add Assignees (of course, you are the assignee by default!)
- [x] DON'T set a Milestone
- [x] DON'T add any labels (PR labels will be added automatically by Jenkins)
- [x] Link to an issue(You can do so by using the required by keyword: required by #5153)
- [x] Add unit tests (or please explain why this PR is not worth having new tests)
- [ ] Pass QA (Sometimes you may encounter an existing flaky test which would obstruct you from getting QA passed. If the failed tests are unrelated, you can add a comment in the PR thread)

### CODE REVIEW

File: ohara-manager/client/src/components/Topic/index.js [hunk 1]
```diff
@@ -14,17 +14,9 @@
  * limitations under the License.
  */
 
-import AddTopicDialog from './AddTopicDialog';
```
Comment (saivirtue) [inline ohara-manager/client/src/components/Topic/index.js:17]: Unused component in current codebase

**Comment by saivirtue** [comment]
retry manager-it

File: ohara-manager/client/cypress/it/navigator.test.ts [hunk 2]
```diff
@@ -0,0 +1,590 @@
...
+    it('should not have navigator after close intro dialog', () => {
+      cy.findByTestId('close-intro-button').click();
+
+      // we should not have navigator since there is no workspace
+      cy.get('#navigator').should('not.exist');
+    });
+  });
+
+  context('Add workspaces', () => {
+    it('should have navigator after add workspace', () => {
```
Comment (konekoya) [inline ohara-manager/client/cypress/it/navigator.test.ts:54]: _adding a workspace_ :)

File: ohara-manager/client/cypress/it/navigator.test.ts [hunk 3]
```diff
@@ -0,0 +1,590 @@
...
+      cy.get('div.shared-topic').within(() => {
+        cy.contains('tr', 'No records to display');
+      });
+      cy.get('div.pipeline-only-topic').within(() => {
+        cy.contains('tr', 'No records to display');
+      });
+    });
+
+    it('should be able to add shared topics', () => {
+      cy.switchSettingSection(SETTING_SECTIONS.topics);
```
Comment (konekoya) [inline ohara-manager/client/cypress/it/navigator.test.ts:124]: Oh wow ❤️

File: ohara-manager/client/cypress/it/navigator.test.ts [hunk 4]
```diff
@@ -0,0 +1,590 @@
...
+
+      // add shared topics
+      cy.findByTitle('Create Topic').should('be.enabled').click();
+
+      cy.findAllByRole('dialog')
+        .filter(':visible')
+        .within(() => {
+          cy.findAllByLabelText('Topic name', { exact: false })
```
Comment (konekoya) [inline ohara-manager/client/cypress/it/navigator.test.ts:132]: I'm a bit confused by this selector... so `cy.findByLabelText('...')` won't work here?

File: ohara-manager/client/cypress/tsconfig.json [hunk 5]
```diff
@@ -1,6 +1,7 @@
 {
   "compilerOptions": {
     "target": "es5",
+    "noEmit": true,
```
Comment (konekoya) [inline ohara-manager/client/cypress/tsconfig.json:4]: 💯

**Comment by konekoya** [comment]
Sorry for the conflicts 😅

File: ohara-manager/client/cypress/it/navigator.test.ts (hunk 4 above)
Comment (saivirtue) [reply ohara-manager/client/cypress/it/navigator.test.ts:132]: The selector would only match the element: `<label>{text}</label>` if not use the option {exact: false} here

Comment (saivirtue) [reply ohara-manager/client/cypress/it/navigator.test.ts:132]: Unfortunately, our `label` element generated by material-ui will have a weird HTML element
```
<label>
"Topic Name"
<span>...</span>
</label>
```

**Comment by saivirtue** [comment]
> Sorry for the conflicts 😅

Not a problem :)

File: ohara-manager/client/cypress/configs/cypress.it.json [hunk 6]
```diff
@@ -11,7 +11,7 @@
   "video": true,
   "videosFolder": "cypress/videos/it",
   "requestTimeout": 10000,
-  "defaultCommandTimeout": 20000,
+  "defaultCommandTimeout": 40000,
```
Comment (saivirtue) [inline ohara-manager/client/cypress/configs/cypress.it.json:14]: Our "custom command" grows bigger and bigger, it needs to relieve the time pressure 😅

File: ohara-manager/client/cypress/support/customCommands.ts [hunk 7]
```diff
@@ -249,3 +266,26 @@ Cypress.Commands.add(
     await connectorApi.remove(connector);
   },
 );
+
+Cypress.Commands.add('deleteAllServices', async () => {
+  await deleteAllServices();
+});
```
Comment (saivirtue) [inline ohara-manager/client/cypress/support/customCommands.ts:272]: This is a "wrap" custom command for us to use native APIs without break the cypress promise-chain

File: ohara-manager/client/src/components/common/Form/Checkbox.js [hunk 8]
```diff
@@ -26,7 +26,7 @@ const InputWrap = styled.div`
 `;
 
 const MuiCheckbox = ({
-  input: { name, onChange, checked = false, ...restInput },
+  input: { name, onChange, checked, ...restInput },
```
Comment (saivirtue) [inline ohara-manager/client/src/components/common/Form/Checkbox.js:29]: Final-form will pass `checked` prop to component of `type=checkbox`, we could rely on it

File: ohara-manager/package.json [hunk 9]
```diff
@@ -12,6 +12,7 @@
     "start": "cross-env NODE_ENV=development nodemon start.js --exec \"yarn lint:server && ts-node\"",
     "start:prod": "cross-env NODE_ENV=production node start.js",
     "start:client": "yarn --cwd client start",
+    "start:client:ci": "yarn --cwd client start:ci",
```
Comment (saivirtue) [inline ohara-manager/package.json:15]: sorry for remove this helper script for local coverage testing 😅

**Comment by saivirtue** [comment]
retry manager-it

**Pull request merged**